### PR TITLE
UICIRC-796: UI tests replacement with RTL/Jest for `FinePolicy` folder in `Models` 

### PR DESCRIPTION
**Pull request by artem-blazhko**
## Purpose

Cover `FinePolicy` folder in `Models`  by RTL/jest tests

## Refs

[UICIRC-796](https://issues.folio.org/browse/UICIRC-796)

## Screenshots

image

image

### CODE REVIEW

File: src/settings/Models/FinePolicy/utils/helper.test.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import helpers from './helpers';
+
+describe('FinePolicy helpers', () => {
+  it('"isValidItemSelected" should return true if some matches', () => {
+    const selectedId = 'selectedId';
+    const options = [
+      { id: 'test' },
+      { id: selectedId },
+    ];
+    expect(helpers.isValidItemSelected(options, selectedId)).toEqual(true);
```
Comment (Stanislau-Snapok) [inline src/settings/Models/FinePolicy/utils/helper.test.js:10]: Could you please add empty line before expect? Here and below.

File: src/settings/Models/FinePolicy/FinePolicy.test.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+  it('"metadata" property should be instance of "Metadata"', () => {
+    expect(finePolicyInstance.metadata).toBeInstanceOf(Metadata);
+  });
+
+  it('"hasValue" should return true if required value more than 0', () => {
+    expect(finePolicyInstance.hasValue('maxOverdueRecallFine')).toEqual(true);
+  });
+
+  it('"hasValue" should return false if required value less than 0', () => {
```
Comment (Stanislau-Snapok) [inline src/settings/Models/FinePolicy/FinePolicy.test.js:63]: less or equal

File: src/settings/Models/FinePolicy/FinePolicy.test.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    expect(finePolicyInstance.hasValue('countClosed')).toEqual(false);
+  });
+
+  it('"hasNonZeroValue" should return true if it is possible to parse value', () => {
+    expect(finePolicyInstance.hasNonZeroValue('maxOverdueRecallFine')).toEqual(true);
+  });
+
+  it('"hasNonZeroValue" should return false if it is not possible to parse value', () => {
```
Comment (Stanislau-Snapok) [inline src/settings/Models/FinePolicy/FinePolicy.test.js:71]: Could you please add one more test to check parsable zero value?

File: src/settings/Models/FinePolicy/FinePolicy.test.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+  it('"hasValue" should return true if required value more than 0', () => {
+    expect(finePolicyInstance.hasValue('maxOverdueRecallFine')).toEqual(true);
+  });
+
+  it('"hasValue" should return false if required value less than 0', () => {
+    expect(finePolicyInstance.hasValue('countClosed')).toEqual(false);
+  });
+
+  it('"hasNonZeroValue" should return true if it is possible to parse value', () => {
```
Comment (Stanislau-Snapok) [inline src/settings/Models/FinePolicy/FinePolicy.test.js:67]: ... to parse value and value not equal 0

File: src/settings/Models/FinePolicy/FinePolicy.test.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+      description: undefined,
+      countClosed: undefined,
+      maxOverdueFine: undefined,
+      forgiveOverdueFine: undefined,
+      gracePeriodRecall: undefined,
+      maxOverdueRecallFine: undefined,
+    }));
+  });
+
+  it('"overdueFine" property should be instance of "OverdueFine"', () => {
```
Comment (Stanislau-Snapok) [inline src/settings/Models/FinePolicy/FinePolicy.test.js:47]: I would suggest to match fields data too. Becaus if someone will drop props passing to the `new OverdueFine()` this test will not fail. We can do mock for class, or in this case, as this class not so complex, we can check it as it is. What do you think?

File: src/settings/Models/FinePolicy/FinePolicy.test.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+    const finePolicy = new FinePolicy({
+      ...policy,
+      overdueFine: {
+        quantity: 1,
+      },
+    });
+    expect(finePolicy.isOverdueFine()).toEqual(true);
+  });
+
+  it('"isOverdueFine" should return false if required value less than 0', () => {
```
Comment (Stanislau-Snapok) [inline src/settings/Models/FinePolicy/FinePolicy.test.js:93]: less or equal. here and below.

File: src/settings/Models/FinePolicy/FinePolicy.test.js [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+  });
+
+  it('"isOverdueFine" should return true if required value more than 0', () => {
+    const finePolicy = new FinePolicy({
+      ...policy,
+      overdueFine: {
+        quantity: 1,
+      },
+    });
+    expect(finePolicy.isOverdueFine()).toEqual(true);
```
Comment (Stanislau-Snapok) [inline src/settings/Models/FinePolicy/FinePolicy.test.js:90]: Add please empty line before expect.

**Review by Stanislau-Snapok — APPROVED** [review]

File: src/settings/Models/FinePolicy/FinePolicy.test.js (hunk 5 above)
Comment (artem-blazhko) [reply src/settings/Models/FinePolicy/FinePolicy.test.js:47]: I am not sure that we need to check it because it depends on properties of `FinePolicy` component and that properies can be empty.
image
 Also, each class like `new OverdueFine()` handle empty properties

File: src/settings/Models/FinePolicy/FinePolicy.test.js (hunk 2 above)
Comment (artem-blazhko) [reply src/settings/Models/FinePolicy/FinePolicy.test.js:63]: done

File: src/settings/Models/FinePolicy/FinePolicy.test.js (hunk 4 above)
Comment (artem-blazhko) [reply src/settings/Models/FinePolicy/FinePolicy.test.js:67]: done

File: src/settings/Models/FinePolicy/FinePolicy.test.js (hunk 3 above)
Comment (artem-blazhko) [reply src/settings/Models/FinePolicy/FinePolicy.test.js:71]: done

File: src/settings/Models/FinePolicy/FinePolicy.test.js (hunk 7 above)
Comment (artem-blazhko) [reply src/settings/Models/FinePolicy/FinePolicy.test.js:90]: done

File: src/settings/Models/FinePolicy/FinePolicy.test.js (hunk 6 above)
Comment (artem-blazhko) [reply src/settings/Models/FinePolicy/FinePolicy.test.js:93]: done

File: src/settings/Models/FinePolicy/utils/helper.test.js (hunk 1 above)
Comment (artem-blazhko) [reply src/settings/Models/FinePolicy/utils/helper.test.js:10]: done

File: src/settings/Models/FinePolicy/FinePolicy.test.js (hunk 5 above)
Comment (Stanislau-Snapok) [reply src/settings/Models/FinePolicy/FinePolicy.test.js:56]: Discussed

File: src/settings/Models/FinePolicy/utils/helper.test.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import helpers from './helpers';
+
+describe('FinePolicy helpers', () => {
```
Comment (Dmitriy-Litvinenko) [inline src/settings/Models/FinePolicy/utils/helper.test.js:3]:
```
describe('FinePolicy helpers', () => {
    const options = [
      { id: 'selector1' },
      { id: 'selector2'},
    ];

  it('"isValidItemSelected" should return true if some matches', () => {
    const selector= 'selector1';

    expect(helpers.isValidItemSelected(options, selector)).toEqual(true);
Stanislau-Snapok marked this conversation as resolved.
  });

  it('"isValidItemSelected" should return false if there is no matches', () => {
    const selector= 'selector3';

    expect(helpers.isValidItemSelected(options, selector)).toEqual(false);
  });
});
```

Comment (Dmitriy-Litvinenko) [reply src/settings/Models/FinePolicy/utils/helper.test.js:3]: Can we move options to reusable part of tests?

**Review by Dmitriy-Litvinenko — APPROVED** [review]

File: src/settings/Models/FinePolicy/utils/helper.test.js (hunk 8 above)
Comment (artem-blazhko) [reply src/settings/Models/FinePolicy/utils/helper.test.js:3]: Fixed
